FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift

```diff
@@ -109,7 +109,13 @@ class AuthBackend: NSObject {
     request.setValue(Bundle.main.bundleIdentifier, forHTTPHeaderField: "X-Ios-Bundle-Identifier")
     request.setValue(requestConfiguration.appID, forHTTPHeaderField: "X-Firebase-GMPID")
     if let heartbeatLogger = requestConfiguration.heartbeatLogger {
-      request.setValue(heartbeatLogger.headerValue(), forHTTPHeaderField: "X-Firebase-Client")
+      if let heartbeatLogger = requestConfiguration.heartbeatLogger {
+        let heartbeatsHeaderValue = await withUnsafeContinuation { continuation in
```

Comment:
That makes sense and seems consistent with other places where we've needed `withUnsafeContinuation`. Good find!

LGTM on green CI

Reply:
So, this didn't work as intended because it never branched off of the original thread because were `await`ing the continuation on the current thread. The trick was to use `async let` to spawn a separate thread. At least from tracing through it, it was evident that the heartbeat work was done off of the thread that it originally was called on here.

I could change this to:
```swift
      async let heartbeatsHeaderValue = await withUnsafeContinuation { continuation in // thread 9
        let heartbeatsHeaderValue = heartbeatLogger.headerValue() // thread 11
        continuation.resume(returning: heartbeatsHeaderValue)
      }
      await request.setValue(heartbeatsHeaderValue, forHTTPHeaderField: "X-Firebase-Client") // thread 9 waiting,
```
but the continuation was unnecessary, so the final result is much cleaner and should ensure the call does not block.